### PR TITLE
Add 5 more health connect sensors

### PR DESCRIPTION
**Pull request by dshokouhi**
## Summary

- Basal body temperature
- Body water mass
- Heart rate variability
- Hydration
- Resting heart rate

## Screenshots

image

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#pendinig

## Any other notes

these are the remaining sensors I plan to add as the rest needs more discussion or we should wait for a request before discussing

all sensors tested with health connect toolbox (side note: v1.9 seems to not insert records properly so rolled back to old build I had downloaded)

### CODE REVIEW

File: app/src/full/java/io/homeassistant/companion/android/sensors/HealthConnectSensorManager.kt
Comment (jpelgrom) [inline app/src/full/java/io/homeassistant/companion/android/sensors/HealthConnectSensorManager.kt:237]: This can be `ml`, no need to capitalize the liter part of it

**Review by jpelgrom — COMMENTED** [review]
Looks like more of the same, which is probably good :)

File: app/src/full/java/io/homeassistant/companion/android/sensors/HealthConnectSensorManager.kt
Comment (dshokouhi) [reply app/src/full/java/io/homeassistant/companion/android/sensors/HealthConnectSensorManager.kt:237]: gemini autocomplete got the best of me there 🙈

**Review by jpelgrom — APPROVED** [review]
Approved assuming that Google also approves the previous batch. Don't forget about documentation

File: common/src/main/res/values/strings.xml
Comment (alfieV) [inline common/src/main/res/values/strings.xml:1342]: there is a small typo here
```suggestion
    <string name="basic_sensor_name_heart_rate_variability">Heart rate variability</string>
```

**Comment by loryanstrant** [comment]
I'm confused, I don't see hydration in the app or in the documentation (https://companion.home-assistant.io/docs/core/sensors/#health-connect-sensors).
Am I missing something?
(I'm running 2025.5.1-full)

**Comment by dshokouhi** [comment]
> I'm confused, I don't see hydration in the app or in the documentation (https://companion.home-assistant.io/docs/core/sensors/#health-connect-sensors). Am I missing something? (I'm running 2025.5.1-full)

merged PRs are not a place for discussion, we had to pull back some sensors due to rejection from Google at the time. We will try to add hydration in the next batch once the current one is approved. There was another thread where we had provided updates on this topic.

**Pull request merged**
